### DISCUSSION TITLE
Harvest at each build vs harvest once and copy data

### DISCUSSION THREAD

**Discussion by Evangelink** [wixtoolset/issues#6445] [Questions]
Hi there,

I am wondering what are the recommandations/guidelines for files declaration. Is it better to do one heat command and then copy and tweak data into the product wxs file or to run the harvest every time and use xsl transformations (or something else) to put things together?

In case there is no real recommendation and it's more on case-by-case basis, what are the pros and cons of each solution?

**Comment by robmen** [comment]
Heat was originally designed to be used once and the results committed to source control. Some wanted to be able to have it run as part of the build process so additional features were bolted on over time. Others choose to specify files explicitly to ensure only the files they want are included in the package, ala "[double-entry bookkeeping](https://en.wikipedia.org/wiki/Double-entry_bookkeeping_system)". Trade-offs are mostly convenience vs correctness.

At FireGiant we developed a more advanced harvesting technology that is designed to be integrated into the build process. You can read about it here: https://www.firegiant.com/wix/wep-documentation/harvesting/
